.eslintrc.js

```diff
@@ -266,7 +266,7 @@ module.exports = {
       },
     },
     {
-      files: ['docs/src/modules/components/**/*.js'],
+      files: ['docs/src/modules/components/**/*{.ts,.tsx,.js,.jsx}'],
```

Comment:
We don't use the .jsx file extension. Aligned https://react.dev/learn#displaying-data or https://nextjs.org/docs/app/building-your-application/routing/defining-routes that defaults to .js

```suggestion
      files: ['docs/src/modules/components/**/*{.ts,.tsx,.js}'],
```

Reply:
But we will use `.tsx` in the future, right? because according to that Next.js [doc](https://nextjs.org/docs/app/building-your-application/routing/defining-routes), `.ts` is not supported

> Good to know: .js, .jsx, or .tsx file extensions can be used for special files.

Personally I'd just always define these four extension in eslint, regardless of whether we use them in certain places or not. Just to avoid that we accidentally stop linting certain files when their extension changes, as this PR uncovered for the `material-ui/no-hardcoded-labels` rule.

Reply:
Correct, we use .tsx

Reply:
> My hope is that this help us to realize .jsx is not supported, more than lining should fail in this case.

Ok, this is also already caught by the [`jsx-filename-extension`](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-filename-extension.md) rule.